src/mdesktopentry.cpp

```diff
@@ -379,11 +383,16 @@ uint MDesktopEntry::hash() const
 
 QString MDesktopEntry::type() const
 {
+    // Type always has to be present (see isValid())
```

Comment:
Being strict, desktop file and isValid() require this, but the API doesn't. But suppose that's irrelevant.